### ISSUE TITLE
Add ability to configure https://pnpm.io/cli/patch-commit --patch-dir via package.json or .npmrc

### ISSUE THREAD

**Issue by Aghassi** [pnpm/pnpm#6508]
- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/pnpm/pnpm/blob/main/CONTRIBUTING.md))

**Describe the user story**

As a developer that maintains a platform in a monorepo of many languages, I don't want the `patches` directory at the root. It's too generic and not relative to other code owners. I would like to configure pnpm to automatically put the patches directory in a specific place when patches occur so I don't have to teach my end user about these flags. Today, a user must know to specify the flag. 
**Describe the solution you'd like**

I'd like the ability to encode this in the `.npmrc` or `package.json` in someway so that I don't have to use `pre` and `post` scripts to move things around and guard against someone putting patches in the wrong place.

**Describe the drawbacks of your solution**

Given this would be opt in, the main drawback is having to dig to find it. The other one is if any packages make assumptions about this directory they would break. I'm not sure drawbacks beyond this.

**Describe alternatives you've considered**

Looking at how `patch-commit` works, I'm trying to add some logic to `postinstall` (which it seems to invoke) which will move patch files from the generated `patches/` dir to the location I care about and then delete the `patches` dir

**Comment by zkochan** [comment]
It is already supported. You can set the `patches-dir` setting.

**Comment by Aghassi** [comment]
I must have missed how to configure that. Can you provide me an example or docs? I was reading the docs yesterday and it wasn't immediately clear how to encode that flag in a config.

**Comment by zkochan** [comment]
All CLI options may be also set through `.npmrc`:

```ini
patches-dir=__patches__
```

**Comment by Aghassi** [comment]
Oh! OK today I learned. Let me try this on Monday. Can I contribute to the docs to make this clearer? Where would you recommend I put it if so? I'm really happy to hear this feature is already supported!

**Comment by Aghassi** [comment]
This works, validated this morning. Thank you so much!

**Comment by zkochan** [comment]
Regarding documentation. I don't know what to do about it. The issue is that all CLI options can be also settings in `.npmc`. But I would prefer not to duplicate the same settings as CLI flags and settings on the npmrc page.

**Comment by Aghassi** [comment]
No need to duplicate, just having a note in either page would be helpful. Something like

> NOTE: All CLI flags are compatible with the `npmrc` so long as they are in the form of flag = > value. If the flag has dashes, convert them to underscores (e.g. flag-with-dashes is flag_with_dashes)

**Comment by zkochan** [comment]
there is no need to convert dashes to underscors. Unless you use env variables via `npm_config_patches_dir`

**Comment by Aghassi** [comment]
You are correct, I mispoke in the how. But to my point, a note on one of the pages would be helpful :). I can contrib it if you'd like
